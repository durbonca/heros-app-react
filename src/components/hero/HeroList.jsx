import { getHeroesByPublisher } from "../../selectors/getHeroByPublisher"

export const HeroList = ( { publisher } ) => {

    const heroes = getHeroesByPublisher(publisher)
    return (
            <ul>
                {heroes.map( hero => { return (
                    <li key={hero.id}>
                        {hero.superhero}
                    </li>)})}
            </ul>
    )
}
