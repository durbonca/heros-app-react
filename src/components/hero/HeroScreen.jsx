import { useParams, Navigate } from 'react-router-dom';
import { getHeroById } from '../../selectors/getHeroById';

export const HeroScreen = () => {

  const { id : heroeId } = useParams();

  const hero = getHeroById(heroeId);  
  if(!hero) {
    return <Navigate to='/' />
  }

  const { superhero, publisher, alter_ego, first_appearance, characters } = hero;

  return (
    <h1>{ superhero }</h1>
  )
}
