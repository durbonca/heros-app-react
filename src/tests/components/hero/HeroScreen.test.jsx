import { Routes, Route, MemoryRouter } from "react-router-dom";
import { mount } from "enzyme";
import { HeroScreen } from "../../../components/hero/HeroScreen";

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockNavigate,
}));

describe('<HeroScreen> tests',()=>{
    
    test("shouldn't show hero if is not hero on URL", () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={['/hero']}>
                <Routes>
                    <Route path="/hero" element={<HeroScreen />} />
                    <Route path="/" element={<h1>No hero page</h1>} />
                </Routes>
            </MemoryRouter>
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('h1').text().trim()).toBe('No hero page');

    })

    test("should to show hero if is hero on URL", () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={['/hero/dc-wonder']}>
                <Routes>
                    <Route path="/hero/:id" element={<HeroScreen />} />
                </Routes>
            </MemoryRouter>
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('h3').text().trim()).toBe('Wonder Woman');

    })

    test("should to return to back screen", () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={['/hero/dc-wonder']}>
                <Routes>
                    <Route path="/hero/:id" element={<HeroScreen />} />
                </Routes>
            </MemoryRouter>
        );
        wrapper.find('button').simulate('click');
        expect(mockNavigate).toHaveBeenCalledWith( -1, { replace: true } ); 
    })

})