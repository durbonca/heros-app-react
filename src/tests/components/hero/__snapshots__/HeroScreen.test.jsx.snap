// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeroScreen> tests should to show hero if is hero on URL 1`] = `
<div
  className="row mt-5"
>
  <div
    className="col-4"
  >
    <img
      alt="dc-wonder"
      className="img-thumbnail animate__animated animate__fadeInLeft"
      src="/assets/dc-wonder.jpg"
    />
  </div>
  <div
    className="col-8 animate__animated animate__fadeIn"
  >
    <h3>
      Wonder Woman
    </h3>
    <ul
      className="list-group list-group-flush"
    >
      <li
        className="list-group-item"
      >
        <b>
          Alter ego:
        </b>
         
        Princess Diana
      </li>
      <li
        className="list-group-item"
      >
        <b>
          Publisher:
        </b>
         
        DC Comics
      </li>
      <li
        className="list-group-item"
      >
        <b>
          First Appearance:
        </b>
         
        All Star Comics #8
      </li>
    </ul>
    <h5
      className="mt-3"
    >
      Characters
    </h5>
    <p>
      Princess Diana
    </p>
    <button
      className="btn btn-outline-info"
      onClick={[Function]}
    >
      Regresar
    </button>
  </div>
</div>
`;

exports[`<HeroScreen> tests shouldn't show hero if is not hero on URL 1`] = `
<h1>
  No hero page
</h1>
`;
